fpdf/image_parsing.py

```diff
@@ -527,13 +532,133 @@ def __getitem__(self, tag):
     return newimgio.read(length)
 
 
+def _to_lzwdata(img, remove_slice=None, select_slice=None):
+    CLEAR_TABLE_MARKER = 256  # Special code to indicate table reset
+    EOD_MARKER = 257  # End-of-data marker
+    INITIAL_BITS_PER_CODE = 9  # Initial code bit width
+    MAX_BITS_PER_CODE = 12  # Maximum code bit width
+
+    data = bytearray(img.tobytes())
+
+    if remove_slice:
+        del data[remove_slice]
+    if select_slice:
+        data = data[select_slice]
+
+    if img.mode == "1":
+        row_size = ceil(img.size[0] / 8)
+        data_with_padding = data
+    else:
+        channels_count = len(data) // (img.size[0] * img.size[1])
+        row_size = img.size[0] * channels_count
+        data_with_padding = bytearray()
+        for i in range(0, len(data), row_size):
+            data_with_padding.extend(b"\0")
+            data_with_padding.extend(data[i : i + row_size])
+
+    data = data_with_padding
```

Comment:
```suggestion
    if img.mode == "1":
        row_size = ceil(img.size[0] / 8)
    else:
        channels_count = len(data) // (img.size[0] * img.size[1])
        row_size = img.size[0] * channels_count
        data_with_padding = bytearray()
        for i in range(0, len(data), row_size):
            data_with_padding.extend(b"\0")
            data_with_padding.extend(data[i : i + row_size])
        data = data_with_padding
```
Is padding only required when `img.mode != "1"`?

In `_to_zdata()` it seems that we always perform padding.

Reply:
I'm not sur that you adressed that comment @opposss? 🙂

---

Could you also please a mention of this addition (the `LZWDecode` compression filter) in `CHANGELOG.md`, as part of this PR?

Reply:
Yes, you're right @Lucas-C , I changed the addition of padding, following the example with `_to_zdata()`, and also added a mention in `CHANGELOG.md` in my last commit.

Reply:
Perfect, good job 👍